### PR TITLE
Update types

### PR DESCRIPTION
**Pull request by stas-demydiuk**
Update TypeScript types usage to the actual types in order to avoid `any` type usage where possible.
Added several overrides for better types handling

### CODE REVIEW

File: build/index.js [hunk 1]
```diff
@@ -36,6 +36,9 @@ let getCurrencyObject = function (currency) {
         }
     }
 };
+function isAmountObject(amount) {
+    return lodash_1.isObject(amount);
```
Comment (wolfen) [inline build/index.js:40]: It would be nice if Fowler's money pattern could be implemented in "clean" Typescript without using lodash at all.

Comment (stas-demydiuk) [reply build/index.js:40]: Also would love that

File: index.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
-import { extend, isFunction, isNaN, isObject, isPlainObject, isString } from 'lodash'
+import { isFunction, isNaN, isObject, isPlainObject, isString } from 'lodash'
```
Comment (macor161) [inline index.ts:1]: Good catch!

File: build/index.js (hunk 1 above)
Comment (macor161) [reply build/index.js:40]: Indeed, I might give it a try in the next few weeks.

**Review by macor161 — APPROVED** [review]
Looking good! 💯 Thank you very much for this pull request. 
I will publish a new release this week.

**Pull request merged**